lib/vdc/DCMPAS.cpp

```diff
@@ -1900,3 +1932,212 @@ bool DCMPAS::DerivedCoordVertFromCell::VariableExists(
 
 	return( _dc->VariableExists(ts, _inName, -1, -1));
 }
+
+//////////////////////////////////////////////////////////////////////
+//
+// Class definitions for derived data variables
+//
+//////////////////////////////////////////////////////////////////////
+
+
+
+DCMPAS::DerivedZonalMeridonal::DerivedZonalMeridonal(
+	string derivedVarName, DC *dc, NetCDFCollection *ncdfc,
+	string normalVarName, string tangentialVarName, bool zonalFlag
+
+) : DerivedDataVar(
+	derivedVarName
+) {
+	_dc = dc;
+	_ncdfc = ncdfc;
+	_normalVarName = normalVarName;
+	_tangentialVarName = tangentialVarName;
+	_zonalFlag = zonalFlag;
+}
+
+int DCMPAS::DerivedZonalMeridonal::Initialize() {
+
+
+	// Set up DataVarInfo for derived variable. No difference between
+	// it and native variables derived from 
+	//
+	bool status = _dc->GetDataVarInfo(_normalVarName, _dataVarInfo);
+	if (! status) {
+		SetErrMsg("Invalid variable \"%s\"", _normalVarName.c_str());
+		return(-1);
+	}
+
+    return(0);
+}
+
+bool DCMPAS::DerivedZonalMeridonal::GetBaseVarInfo(
+	DC::BaseVar &var
+) const {
+	var = _dataVarInfo;
+	return(true);
+}
+
+bool DCMPAS::DerivedZonalMeridonal::GetDataVarInfo(
+	DC::DataVar &cvar
+) const {
+	cvar = _dataVarInfo;
+	return(true);
+}
+
+int DCMPAS::DerivedZonalMeridonal::GetDimLensAtLevel(
+    int , std::vector <size_t> &dims_at_level,
+    std::vector <size_t> &bs_at_level
+) const {
+	dims_at_level.clear();
+	bs_at_level.clear();
+
+	int rc = _dc->GetDimLensAtLevel(_normalVarName, -1, dims_at_level,bs_at_level);
+	if (rc<0) return(-1);
+
+	// Never blocked
+	//
+    bs_at_level = vector <size_t> (dims_at_level.size(), 1);
+
+	return(0);
+}
+
+int DCMPAS::DerivedZonalMeridonal::OpenVariableRead(
+    size_t ts, int , int 
+) {
+
+	DC::FileTable::FileObject *f = new DC::FileTable::FileObject(
+		ts, _derivedVarName, 0, 0, 0
+	);
+
+	return(_fileTable.AddEntry(f));
+}
+
+int DCMPAS::DerivedZonalMeridonal::CloseVariable(int fd) {
+	DC::FileTable::FileObject *f = _fileTable.GetEntry(fd);
+
+	if (! f) {
+		SetErrMsg("Invalid file descriptor : %d", fd);
+		return(-1);
+	}
+
+	_fileTable.RemoveEntry(fd);
+	delete f;
+
+	return(0);
+}
+
+
+
+int DCMPAS::DerivedZonalMeridonal::ReadRegion(
+	int fd,
+    const vector <size_t> &min, const vector <size_t> &max, float *region
+) {
+	VAssert(min.size() == 2);
+	VAssert(min.size() == max.size());
+
+	DC::FileTable::FileObject *f = _fileTable.GetEntry(fd);
```

Comment:
Line 2016 checks `_fileTable.GetEntry()` returning a nullptr, while this line doesn't check it. Is this check necessary?

Reply:
Indeed. I've fixed it.